### ISSUE TITLE
get_password_data fails for Windows machine that has already changed the password

### ISSUE THREAD

**Issue by alexmanly** [chef/chef-provisioning-aws#320]
If I try to create a machine from a golden image and the image is a Windows box with a pre-set password, then provisioning fails to get the password data from AWS.  get_password_data fails to return any data.  

You should be able to specify the windows administrator password in the recipe and that password should be used to connect to the machine once it is running.

```ruby
machine chefdk do
    machine_options bootstrap_options: {
      image_id: node['chef-aws-demo-infrastructure']['chef-dk-ami'],
      instance_type: 'c4.xlarge',
      private_ip_address: '10.0.0.60',
      key_name: "#{key}",
      subnet_id: aws_subnet_name,
      security_group_ids: aws_security_group_name,
      associate_public_ip_address: true
    },
    :is_windows => true,
    :ssh_username => "Administrator",
    :password => 'Cod3Can!'
    tag "#{key} #{name} Demo ChefDK Workstation"
  end
```

This is the chef client run output that ultimately fails.

```bash
[2015-09-10T12:44:29+01:00] INFO: 510/900s elapsed -- sleeping 10 for Customer-ABC-chef-dk's admin password.
[2015-09-10T12:44:40+01:00] INFO: [AWS EC2 200 0.532331 0 retries] get_password_data(:instance_id=>"i-ed63d64d")

[2015-09-10T12:44:40+01:00] INFO: 520/900s elapsed -- sleeping 10 for Customer-ABC-chef-dk's admin password.
[2015-09-10T12:44:50+01:00] INFO: [AWS EC2 200 0.542405 0 retries] get_password_data(:instance_id=>"i-ed63d64d")

[2015-09-10T12:44:50+01:00] INFO: 530/900s elapsed -- sleeping 10 for Customer-ABC-chef-dk's admin password.
[2015-09-10T12:45:01+01:00] INFO: [AWS EC2 200 0.507712 0 retries] get_password_data(:instance_id=>"i-ed63d64d")
```

**Comment by alexmanly** [comment]
Found out this is due to the fact I have used an AMI with a Windows username and password pre-configured.   Once I configured the machine to restart using sysprep and the setting EC2 set the random password, the next error was the Chef powershell script to install Chef was too long for WinRM.  It is related to this bug: https://github.com/chef/chef-provisioning/issues/431

**Comment by thommay** [comment]
Don't think this should be closed; supplying a password seems like a legit thing to do (and, down the road, we probably want to support windows nodes that are joined to a domain in AWS, which'll have their own set of weirdnesses but regardless probably don't support the AWS password service.)

**Comment by tyler-ball** [comment]
Yeah, I think this is a legit bug.  We have some code in kitchen-ec2 that correctly handles this use case - we should look at that.

**Comment by hh** [comment]
I'm running into this as well, if you where to use machine_image to capture an AMI, the password would not be reset when using that ami as the base for a new one, so we'd need to provide it OR reset it.

I'm trying to see if it's possible to  force regeneration of a password if we do not provide one.

So far I've found: https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2-windows-passwords.html#ResettingAdminPassword

Modify the configuration file on the secondary volume as follows:

From the temporary instance, open \Program Files\Amazon\Ec2ConfigService\Settings\config.xml using a text editor, such as Notepad.

At the top of the file, find the plugin with the name Ec2SetPassword, as shown here. Change the state from Disabled to Enabled and then save the file.

**Comment by hh** [comment]
@alexmanly  How did you configured the machine to restart using sysprep and the setting EC2 set the random password? That seem to be the only route to use existing custom AMIs until this bug is fixed.

**Comment by robcoward** [comment]
While we don't yet use machine_image to generate AMI's, we do use packer with a chef-client provisioner and have the following resource in a recipe:
```ruby
powershell_script 'ConfigureEc2ConfigService' do
  code <<-EOH
$EC2SettingsFile="C:\Program Files\Amazon\Ec2ConfigService\Settings\Config.xml"
$xml = [xml](get-content $EC2SettingsFile)
$xmlElement = $xml.get_DocumentElement()
$xmlElementToModify = $xmlElement.Plugins
foreach ($element in $xmlElementToModify.Plugin)
{		
	if ($element.name -eq "Ec2HandleUserData")
	{
		$element.State="Enabled"
	}
	if ($element.name -eq "Ec2SetPassword")
	{
		$element.State="Enabled"
	}}	
$xml.Save($EC2SettingsFile)
  EOH
end
```
Hope that helps.

**Comment by alexmanly** [comment]
@hh I did this manually :(.  Using the instructions here: http://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/UsingConfig_WinAMI.html

From the Start menu, click All Programs, and then click EC2ConfigService Settings, then on the image tab, select Random password and then "Shutdown with Sysprep"

**Comment by hh** [comment]
@alexmanly yea I see that now.. set to Random:

![image](https://cloud.githubusercontent.com/assets/31331/9914668/e8aa6872-5c80-11e5-9f88-e3fe3761db9a.png)

**Comment by hh** [comment]
@robcoward that looks great

**Comment by cgunasree08** [comment]
This issue or pull request is being automatically closed because the repository is being archived as part of the [Repository Standardization Initiative](https://github.com/chef-boneyard/oss-repo-standardization-2025).
